nexus/src/app/background/init.rs

```diff
@@ -39,128 +122,208 @@ use oximeter::types::ProducerRegistry;
-    pub external_endpoints: tokio::sync::watch::Receiver<
-        Option<external_endpoints::ExternalEndpoints>,
-    >,
-    /// task handle for the ipv4 nat entry garbage collector
-    pub task_nat_cleanup: TaskHandle,
-
-    /// task handle for the switch bfd manager
-    pub task_bfd_manager: TaskHandle,
-
-    /// task handle for the task that collects inventory
-    pub task_inventory_collection: TaskHandle,
-
-    /// task handle for the task that collects inventory
-    pub task_physical_disk_adoption: TaskHandle,
-
-    /// task handle for the task that detects phantom disks
-    pub task_phantom_disks: TaskHandle,
-
-    /// task handle for blueprint target loader
-    pub task_blueprint_loader: TaskHandle,
-
-    /// task handle for blueprint execution background task
-    pub task_blueprint_executor: TaskHandle,
-
-    /// task handle for collecting CockroachDB node IDs
-    pub task_crdb_node_id_collector: TaskHandle,
-
-    /// task handle for the service zone nat tracker
-    pub task_service_zone_nat_tracker: TaskHandle,
-
-    /// task handle for the switch port settings manager
-    pub task_switch_port_settings_manager: TaskHandle,
-
-    /// task handle for the opte v2p manager
-    pub task_v2p_manager: TaskHandle,
-
-    /// task handle for the task that detects if regions need replacement and
-    /// begins the process
-    pub task_region_replacement: TaskHandle,
+    // Handles to activate specific background tasks
+    pub task_internal_dns_config: Activator,
+    pub task_internal_dns_servers: Activator,
+    pub task_external_dns_config: Activator,
+    pub task_external_dns_servers: Activator,
+    pub task_metrics_producer_gc: Activator,
+    pub task_external_endpoints: Activator,
+    pub task_nat_cleanup: Activator,
+    pub task_bfd_manager: Activator,
+    pub task_inventory_collection: Activator,
+    pub task_physical_disk_adoption: Activator,
+    pub task_phantom_disks: Activator,
+    pub task_blueprint_loader: Activator,
+    pub task_blueprint_executor: Activator,
+    pub task_crdb_node_id_collector: Activator,
+    pub task_service_zone_nat_tracker: Activator,
+    pub task_switch_port_settings_manager: Activator,
+    pub task_v2p_manager: Activator,
+    pub task_region_replacement: Activator,
+    pub task_region_replacement_driver: Activator,
+    pub task_instance_watcher: Activator,
+    pub task_service_firewall_propagation: Activator,
+    pub task_abandoned_vmm_reaper: Activator,
+    pub task_vpc_route_manager: Activator,
+
+    // Handles to activate background tasks that do not get used by Nexus
+    // at-large.  These background tasks are implementation details as far as
+    // the rest of Nexus is concerned.  These handles don't even really need to
+    // be here, but it's convenient.
+    task_internal_dns_propagation: Activator,
+    task_external_dns_propagation: Activator,
+
+    // Data exposed by various background tasks to the rest of Nexus
```

Comment:
looks like a typo:
```suggestion
    /// Data exposed by various background tasks to the rest of Nexus
```

Reply:
Heh, it's not: there are three categories of fields in this struct and non-doc comments describe each section.  Right now there's only one thing in this bucket (but I expect we'll have more).